File: python/longestpalindromicsubstring.py
class Solution(object):
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        self.lo = 0
        self.maxLen = 0
        if len(s) < 2:
            return s
        for i in xrange(len(s)):
            self.maxPalindrome(i,i,s)
            self.maxPalindrome(i, i+1, s)
            
        return s[self.lo: self.lo+self.maxLen]
        
    def maxPalindrome(self, i, j, s):
        while i >= 0 and j < len(s) and s[i] == s[j]:
            i -= 1
            j += 1
        if self.maxLen < j-i-1:
            self.lo = i+1
            self.maxLen = j-i-1

File: java/longestPalindromicSubstring.java
class Solution {
    private int index, maxLen;
    public String longestPalindrome(String s) {
        int len = s.length();
	    if (len < 2)
		    return s;
        for (int i = 0; i < s.length() - 1; i++) {
            LPS(s, i, i);
            LPS(s, i, i + 1);
        }
        return s.substring(index, index + maxLen);
    }
    
    private void LPS(String s, int i, int j) {
        while (i >= 0 && j < s.length() && s.charAt(i) == s.charAt(j)) {
            i--;
            j++;
        }
        
        if (maxLen < j - i - 1) {
            maxLen = j - i - 1;
            index = i + 1;
        }
    }
}
